File: Lab 2/MATLAB/Task_4.m
% Ring Tone

fs2 = 10000;
T2 = 23;
t2 = 0:1/fs2:T2;

fr1 = 440;
fr2 = 480;

ring_tone = 0.1*sin(2*pi*fr1*t2) + 0.1*sin(2*pi*fr2*t2);


% Loop to set the Cadence for 2 seconds ON and then 4 seconds OFF 
for i = 1:1:T2
    if((mod(i,6) == 2)||(mod(i,6) == 3)||(mod(i,6) == 4)||(mod(i,6) == 5))
        ring_tone(1,i*fs2:((i+1)*fs2-1)) = 0;
    else 
        ring_tone(1,i*fs2:((i+1)*fs2-1)) = ring_tone(1,i*fs2:((i+1)*fs2-1));
    end
end

sound(ring_tone,fs2)

filename = 'ring_tone_983.wav';
audiowrite(filename,ring_tone,fs2)

ring_tone = ring_tone(1,1:numel(t2));
figure(1)
plot(t2,ring_tone)
title("Time Domain Plot - Busy Tone")
xlabel("Time(s)")
ylabel("Amplitude")
axis([0 10 -0.3 0.3]) 

File: Lab 2/Python/Task_4.py
import numpy as np
import math
import matplotlib.pyplot as plt
from scipy.io import wavfile
from playsound import playsound
import os

# Dial tone

fs1 = 44100
t_start = 0
t_stop = 0.5
ts = 1/fs1
t1 = np.arange(t_start, t_stop, ts)

fd1 = 440
fd2 = 350

dial_tone = 0.1*np.sin(2*math.pi*fd1*t1) + 0.1*np.sin(2*math.pi*fd2*t1)

dir_name = os.getcwd()
dir_name_new = '{}'.format(dir_name).replace('\\', '/')
filename = "dial_tone.wav"
dir_name_final = dir_name_new + '/' + filename
print(dir_name_final)
wavfile.write(filename, fs1, dial_tone)
playsound(dir_name_final)

plt.figure(1)
plt.plot(t1, dial_tone)
plt.title("Time Domain Plot - Dial Tone")
plt.xlabel("Time(s)")
plt.ylabel("Amplitude")

axes = plt.gca()
axes.set_xlim([0, 0.5])
axes.set_ylim([-0.3, 0.3])
plt.show()
